### PR TITLE
feat: Enable login button when correct email is entered

### LINKED ISSUE #522: Dont enable login button until and unless correct format of email is entered
working  on it 👍

### PR DESCRIPTION
**Pull request by dreadpool2**
fixes #522

### CODE REVIEW

**Comment by dreadpool2** [comment]
Message updated

**Comment by simarsingh24** [comment]
why is login branch added to fossasia's open event repo? this has been told you many times! Please make sure this never happens again

**Comment by simarsingh24** [comment]
@dreadpool2 delete that branch too

**Comment by dreadpool2** [comment]
@simarsingh24 Branch deleted

**Comment by nikit19** [comment]
@dreadpool2 Please use smaller commit messages.
You can use the description to describe what's happening in the PR

**Comment by dreadpool2** [comment]
@nikit19 Tried to shorten the message as much as possible

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 1]
```diff
@@ -75,10 +75,12 @@ class LoginFragment : Fragment() {
             override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
 
             override fun onTextChanged(email: CharSequence, start: Int, before: Int, count: Int) {
-                if (loginActivityViewModel.showForgotPassword(email.toString())) {
+                if (loginActivityViewModel.isCorrectEmail(email.toString())) {
                     rootView.forgotPassword.visibility = View.VISIBLE
+                    rootView.loginButton.isEnabled = true
                 } else {
                     rootView.forgotPassword.visibility = View.GONE
+                    rootView.loginButton.isEnabled = false
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Bind to LiveData

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Updated

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 2]
```diff
@@ -114,4 +116,9 @@ class LoginFragment : Fragment() {
         activity?.overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_right)
         activity?.finish()
     }
+
+    private fun onEmailEntered(enable: Boolean) {
+        rootView.loginButton.isEnabled = enable
+        if (enable) rootView.forgotPassword.visibility = View.VISIBLE else rootView.forgotPassword.visibility = View.GONE
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: `rootView.forgotPassword.visibility = if (enable) View.VISIBLE else View.GONE`

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Updated

**Pull request merged**